modules/core/src/main/java/org/apache/ignite/internal/processors/cache/ValidationOnNodeJoinUtils.java

```diff
@@ -209,6 +203,44 @@
         return null;
     }
 
+    /**
+     * @param locDesc Local cache descriptor.
+     * @param joinedData Joined node cache data.
+     * @param isGridActive Is grid active.
+     * @param errorMsg Error message builder.
+     */
+    static void validateSqlSchema(
+        DynamicCacheDescriptor locDesc,
+        StoredCacheData joinedData,
+        boolean isGridActive,
+        StringBuilder errorMsg
+    ) {
+        String joinedSchema = joinedData.config().getSqlSchema();
+        String locSchema = locDesc.cacheConfiguration().getSqlSchema();
+
+        // Peform checks of SQL schema. If schemas' names not equal, only valid case is if local or joined
+        // QuerySchema is empty and schema name is null (when indexing enabled dynamically).
+        if (!F.eq(joinedSchema, locSchema)
+            && (locSchema != null || !locDesc.schema().isEmpty())
+            && (joinedSchema != null || !F.isEmpty(joinedData.queryEntities()))
+        )
+            errorMsg.append(String.format(SQL_SCHEMA_CONFLICTS_MESSAGE, locDesc.cacheName(), joinedSchema,
```

Comment:
Typo: Brackets for multiline if required.

Reply:
Do we have a test to check this?

Reply:
Yes, please see IgniteDynamicEnableIndexingRestoreTest